c/meterpreter/source/extensions/stdapi/server/sys/registry/registry.c

```diff
@@ -35,16 +31,11 @@ DWORD request_registry_check_key_exists(Remote *remote, Packet *packet)
 		packet_add_tlv_bool(response, TLV_TYPE_BOOL, exists);
 		result = ERROR_SUCCESS;
 	}
-	else
-	{
-		dprintf("[REG] Invalid parameter");
-		result = ERROR_INVALID_PARAMETER;
-	}
 
-	dprintf("[REG] Returning result: %u %x", result, result);
-	packet_transmit_response(result, remote, response);
+	free(baseKey);
 
-	dprintf("[REG] done.");
+	packet_transmit_response(result, remote, response);
+out:

```

Comment:
Returning failure from this function will shut down the session entirely. So, in general, all meterpreter functions return ERROR_SUCCESS to the packet dispatcher, even if they transmit a failure packet.

Of course, if we do really get a memory allocation failure, I suspect we'll probably crash soon anyway.